openfisca_core/tests/assets/variable_parameters.yaml

```diff
@@ -0,0 +1,99 @@
+# Use the VAR node and its condition children
+# to avoid using variations of parameter names to express conditions,
+# and compute the condition in your formula.
+
+# This is very useful for "terminal variations",
+# that is variations that happen close to the leaf nodes of the param tree
+
+# conditions use real variable names that you will pass from your formula
+# TODO or should we compute them for you ? Ideally yes, but what about period conversions ?
+
+# Regular value parameters :
+
+- variable: fillon_taux_max
+  description: Taux maximum de l'allègement général
+  format: percent
+  VAR:
+    - condition: "20 <= effectif_entreprise"
+      VALUES:
+        2018-01-01: .2850
+        2017-01-01: .2850
+        2016-01-01: .2842
+        2015-01-01: .2835
+        2005-07-01: .26
+    - condition: "effectif_entreprise < 20"
```

Comment:
I really like the idea of declarative conditions, but I'm afraid it will lead to “monsters”: conditions that are too complex for proper declarative format, and that are partly defined in the parameter table, and partly in the code.

To evaluate this risk, we could try to find conditions that can not be expressed in this format. If none is found in the current codebase, the risk is pretty low.

Otherwise, one alternative could be to name the value sets, and leave it to the code to access the different value sets. This would remind the current behaviour though.